src/ReportManager/SchemaFormContext/SchemaForm/index.js

```diff
@@ -0,0 +1,70 @@
+import React, { forwardRef, useState } from 'react';
+
+import SchemaFormContextProvider from './SchemaFormContext';
+import Section from './fields/Section';
+
+import styles from './styles.module.scss';
+import { FORM_FIELDS_TYPES } from '../../constants';
+import { TextFieldValidators } from './fields/Text';
+import { getSchemaFieldUIType, isFieldActive } from './SchemaFormContext/utils';
+
+const FormFieldValidators = {
+  [FORM_FIELDS_TYPES.TEXT]: TextFieldValidators
+};
+
+const SchemaForm = ({ schema, formData = {}, onFormSubmit, className, onChange }, ref) => {
```

Comment:
This forwardedRef doesn't follow the classic pattern of being a ref injected to the whole component. It is being injected just to the button, and there's no way the parent can know this explicitly. An alternative to make what that ref is explicit is to use a renderProp:
```
const Parent = () => {
  const submissionButtonRef = useRef();

  return <SchemaForm renderSubmissionButton={() => <button ref={submissionButtonRef} ... />} />;
};

const SchemaForm = ({ renderSubmissionButton }) => {
  ...
  return <form onSubmit={handleOnSubmit} className={className}>
    {
      schema?.ui?.order.map((sectionName) => <Section sectionName={sectionName} key={sectionName} />)
    }
    {renderSubmissionButton()}
  </form>;
};
```

Now you have the ref in the parent, where you can hide the button and trigger the submission of the form programmatically!

Reply:
Got it, I did not put much attention when I was trying to homologate the new form behavior with the old one, but yeah, this approach makes way more sense, thanks 🫡